Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -213,6 +215,50 @@ public final class CoreDataManager: StorageManagerType {
             return debugMessages
         }
     }
+
+    /// Note that we have to enumerate all the files in our sandbox to get these properties, so this takes some time.
+    /// It is intended _only_ for use when logging a crash, or other high-value log which won't get in the user's way.
+    private static func storageSizeLogProperties() -> [String: Int64]? {
+        let directoryUrls = [
+            "Bundle": Bundle.main.bundleURL,
+            "Documents": FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!,
+            "Library": FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask).first!,
+            "Temp": FileManager.default.temporaryDirectory,
+        ]
+
+        let directorySizes = directoryUrls.mapValues { url in
+            getAllocatedStorageSize(for: url)
+        }
+
+        let totalSize = directorySizes.values.reduce(0) { partialResult, size in
+            return partialResult + size
+        }
```

Comment:
nit;

reduce(into:) i[s theoretically more performant](https://developer.apple.com/documentation/swift/array/reduce(into:_:)#discussion) for operations such as these, although I didn't measure if it gives any improvement in this case:

```suggestion
        let totalSize = directorySizes.values.reduce(into: 0) { result, size in
            result += size
        }
```

Reply:
I think that's only more performant where the `into` is a collection – in this case, we're reducing into an Int so this is the same/better performance wise...

> This method is preferred over reduce(_:_:) for efficiency when **the result** is a copy-on-write type, for example an Array or a Dictionary.

(emphasis mine)

Reply:
Thanks for clarifying!